gensound/transforms.py

```diff
@@ -188,33 +188,54 @@ def realise(self, audio):
 ######### Level/ampltidue Stuff ###################
 
 class Fade(Transform):
-    min_fade = -50
-    
-    def __init__(self, is_in=True, duration=3e3):
+    """ Adds Fade In / Out to the signal with different curve presets
+
+    """
+    def __init__(self, is_in=True, duration=3e3, curve="linear",  degree=1):        
+        assert curve in ("linear", "polynomial")
         self.is_in = is_in
         self.duration = duration
-    
+        self.curve = curve
+        self.degree = degree
+
     def realise(self, audio):
-        amp = DB_to_Linear(np.linspace(Fade.min_fade, 0, self.num_samples(audio.sample_rate)))
-        #amp = np.linspace(0, 1, self.num_samples(audio.sample_rate))
-        # perhaps the fade in should be nonlinear
-        # TODO subsciprability problem
+        if self.curve == "linear":
+            amp = np.linspace(0, 1, self.num_samples(audio.sample_rate))
+        elif self.curve == "polynomial":
+            amp = (np.linspace(0, 1, self.num_samples(audio.sample_rate))) ** self.degree      
         
-        if self.is_in:
+        # fade in/out handler
+        if self.is_in: 
             audio.audio[:,:len(amp)] *= amp
-        else:
-            audio.audio[:,-len(amp):] *= amp[::-1]
+        else: 
+            audio.audio[:,-len(amp):] *=  amp[::-1]
+
```

Comment:
These lines haven't changed actually, accept for an additional comment, why does this show up as part of the diff?

Reply:
It was one-liners - and now with line breaks.

Reply:
Yes, github showed it like the lines were replaced with identical lines. Now it's fine, except for the redundant space on line 211